src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -8,7 +8,7 @@
  *
  * @internal
  */
-class BlockTemplatesCompatibility {
+class BlockTemplatesCompatibility extends AbstractTemplateCompatibility {
```

Comment:
We should rename this class to remind more about what it does, like `ArchiveProductTemplateCompatibility`. But still, the best case IMO is we can use one class for both template types.

Reply:
I agree with you! I didn't rename it in this PR to avoid confusion. I will create another PR to rename this class when we merge this PR.